flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -225,16 +226,38 @@ CompletableFuture<TransientBlobKey> requestTaskManagerFileUploadByType(
 
     /**
      * Request the file upload from the given {@link TaskExecutor} to the cluster's {@link
-     * BlobServer}. The corresponding {@link TransientBlobKey} is returned.
+     * BlobServer}. The corresponding {@link TransientBlobKey} is returned. To support different
+     * type file upload with name, using {@link
+     * ResourceManager#requestTaskManagerFileUploadByNameAndType} as instead.
      *
      * @param taskManagerId identifying the {@link TaskExecutor} to upload the specified file
      * @param fileName name of the file to upload
      * @param timeout for the asynchronous operation
      * @return Future which is completed with the {@link TransientBlobKey} after uploading the file
      *     to the {@link BlobServer}.
+     * @deprecated use {@link #requestTaskManagerFileUploadByNameAndType(ResourceID, String,
+     *     FileType, Duration)} as instead.
      */
+    @Deprecated
     CompletableFuture<TransientBlobKey> requestTaskManagerFileUploadByName(
-            ResourceID taskManagerId, String fileName, @RpcTimeout Time timeout);
+            ResourceID taskManagerId, String fileName, @RpcTimeout Duration timeout);
```

Comment:
We do not need to change the class from `Time` to `Duration` for a deprecated interface.